### PR TITLE
filetype: linker map files cannot be recognized

### PR DESCRIPTION
**Pull request by Freed-Wu**
Problem: linker map files cannot be recognized
Solution: Recognize linker map files as linkermap filetype

### CODE REVIEW

**Comment by Freed-Wu** [comment]
https://github.com/MapServer/MapServer-documentation/issues/990

vim syntax for mapserver's map file is missing.

**Comment by Freed-Wu** [comment]
example:

```markdown
******************************************************************************
               TMS320C6x Linker Unix v7.4.24                   
******************************************************************************
>> Linked Sat Dec 28 20:51:14 2024

OUTPUT FILE NAME:   <x264.out>
ENTRY POINT SYMBOL: "_c_int00"  address: 8204eb20


MEMORY CONFIGURATION

         name            origin    length      used     unused   attr    fill
----------------------  --------  ---------  --------  --------  ----  --------
  ARM_IRAM              00000000   00008000  00000000  00008000  RWIX
  ARM_IROM              00008000   00008000  00000000  00008000  RWIX
  ARM_DRAM              00010000   00008000  00000000  00008000  RWIX
  ARM_DROM              00018000   00008000  00000000  00008000  RWIX
  SH_DSP_L2             11818000   00020000  00000000  00020000  RWIX
  SH_DSP_L1P            11e00000   00008000  00000000  00008000  RWIX
  SH_DSP_L1D            11f00000   00008000  00000000  00008000  RWIX
  PCI_DATA              30000000   10000000  00000000  10000000  RWIX
  EMIFA_CS2             42000000   02000000  00000000  02000000  RWIX
  EMIFA_CS3             44000000   02000000  00000000  02000000  RWIX
  EMIFA_CS4             46000000   02000000  00000000  02000000  RWIX
  EMIFA_CS5             48000000   02000000  00000000  02000000  RWIX
  VLYNQ_DATA            4c000000   04000000  00000000  04000000  RWIX
  DDR2                  80000000   20000000  020a36cb  1df5c935  RWIX


SECTION ALLOCATION MAP

 output                                  attributes/
section   page    origin      length       input sections
--------  ----  ----------  ----------   ----------------
.pinit     0    80000000    00000000     UNINITIALIZED

.data      0    80000000    00000000     UNINITIALIZED

.stack     0    80000000    01000000     UNINITIALIZED
                  80000000    00000008     rts64plus.lib : boot.obj (.stack)
                  80000008    00fffff8     --HOLE--

.sysmem    0    81000000    01000000     UNINITIALIZED
                  81000000    00000008     rts64plus.lib : memory.obj (.sysmem)
                  81000008    00fffff8     --HOLE--



GLOBAL SYMBOLS: SORTED ALPHABETICALLY BY Name 

address    name
--------   ----
820a36a4   $bss
820a36a4   .bss
80000000   .data
82000000   .text
8204faa0   C$$EXIT


GLOBAL SYMBOLS: SORTED BY Symbol Address 

address    name
--------   ----
00000001   __TI_args_main
01000000   __STACK_SIZE
01000000   __SYSMEM_SIZE

[453 symbols]


SYMBOL DEFINITIONS BY INPUT FILE 

================================================================================

downsample.c.o

Run/Load            
Value     Binding   Name (Section)
--------  --------  ---------------
000004ec  local     $C$DW$L$_resize$2$B (.text)
00000524  local     $C$DW$L$_resize$2$E (.text)



================================================================================

rts64plus.lib : strtoull.obj

Run/Load            
Value     Binding   Name (Section)
--------  --------  ---------------
8203f3ec  local     $C$L10 (.text:_strtoull)
8203f41c  local     $C$L11 (.text:_strtoull)


================================================================================

<linker>

<< No Symbols Defined >>


================================================================================

```

**Comment by Freed-Wu** [comment]
Screenshot_20241228_214211

**Comment by chrisbra** [comment]
Thanks. You removed a test, can you add a proper test for the `map/lnkmap` filetype then please?

File: src/testdir/test_filetype.vim [hunk 1]
```diff
@@ -2774,6 +2773,24 @@ func Test_make_file()
   filetype off
 endfunc
 
+func Test_map_file()
+  filetype on
+
+  " TI linker map file
+  call writefile(['******************************************************************************', '               TMS320C6x Linker Unix v7.4.24                   ', '******************************************************************************'], 'Xfile.map', 'D')
+  split Xfile.map
+  call assert_equal(1, get(b:, 'lnkmap', 0))
```
Comment (chrisbra) [inline src/testdir/test_filetype.vim]:
```suggestion
  call assert_equal('lnkmap', &filetype)
```

File: src/testdir/test_filetype.vim [hunk 2]
```diff
@@ -2774,6 +2773,24 @@ func Test_make_file()
...
+  bwipe!
+
+  " TI linker map file
+  call writefile(['MAP', 'NAME "local-demo"', 'END'], 'Xfile.map', 'D')
+  split Xfile.map
+  call assert_equal(1, get(b:, 'map', 0))
```
Comment (chrisbra) [inline src/testdir/test_filetype.vim]:
```suggestion
  call assert_equal('map', &filetype)
```

File: runtime/ftplugin/lnkmap.vim [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+if exists("b:did_ftplugin")
+  finish
+endif
+
+" Don't load another plugin for this buffer
+let b:did_ftplugin = 1
+
+" Using line continuation here.
+let s:cpo_save = &cpo
+set cpo-=C
```
Comment (chrisbra) [inline runtime/ftplugin/lnkmap.vim]: you are not using line-continuation here, so you can remove the `s:cpo_save`

File: runtime/ftplugin/lnkmap.vim [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+let b:undo_ftplugin = "setl iskeyword"
```
Comment (chrisbra) [inline runtime/ftplugin/lnkmap.vim]:
```suggestion
let b:undo_ftplugin = "setl iskeyword<"
```

File: runtime/ftplugin/lnkmap.vim [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+set iskeyword+=.
```
Comment (chrisbra) [inline runtime/ftplugin/lnkmap.vim]:
```suggestion
setl iskeyword+=.
```

**Comment by chrisbra** [comment]
thanks

File: src/testdir/test_filetype.vim [hunk 6]
```diff
@@ -2774,6 +2773,24 @@ func Test_make_file()
...
+func Test_map_file()
+  filetype on
+
+  " TI linker map file
+  call writefile(['******************************************************************************', '               TMS320C6x Linker Unix v7.4.24                   ', '******************************************************************************'], 'Xfile.map', 'D')
+  split Xfile.map
+  call assert_equal('lnkmap', &filetype)
+  bwipe!
+
+  " TI linker map file
```
Comment (zeertzjq) [inline src/testdir/test_filetype.vim:2785]: Shouldn't it say `UMN mapserver config file` here?

Comment (Freed-Wu) [reply src/testdir/test_filetype.vim:2785]: Sorry for a typo.

Comment (Freed-Wu) [reply src/testdir/test_filetype.vim:2785]: Fixed here #16320

**Pull request closed without merging**